### DISCUSSION TITLE
/help slash-command

### DISCUSSION THREAD

**Discussion by S1M0N38** [olimorris/codecompanion.nvim#207] [Ideas]
Neovim and high-quality plugins are shipped with well-structured docs in doc/*.txt.
This documentation is indexed with tags and easily searchable (e.g., `Telescope help_tags`).

I think that a useful slash-command could be `/help` which triggers something like `Telescope help_tags` and inserts the selected documentation section in that chat.

---

Another approach is the one used by VSCode: it uses `@`-commands to provide context across multiple files. Right now, three `@`-commands are implemented:
- `@workspace`
- `@terminal`
- `@vscode`

For example, the prompt
```
@vscode how can I change the text font size?
```
will use the VSCode docs to perform RAG.

However, I think that this second option (e.g., `@neovim`) is not straightforward and it would only make sense to reconsider if embeddings (and how to use them) will be supported in the future by codecompanion.nvim

**Comment by olimorris** [comment]
This is an awesome suggestion.

I'm currently reworking Tools. These allow you to `@` a particular tool into the chat buffer which gives the LLM the ability to execute commands/functions.

A help tool would be a brilliant idea. We could share a list of help docs available to the LLM and let it decide what it would like to consume.

I do have a `@rag` tool in the plugin right now which leverages a third party service to extract data from a page for an LLM but it needs a little bit of work.

My main focus is to add a tool which can update a buffer next if a user gives approval.
